process/transactionEvaluator/transactionEvaluator.go

```diff
@@ -91,7 +98,9 @@ func (ate *apiTransactionEvaluator) SimulateTransactionExecution(tx *transaction
 		ate.mutExecution.Unlock()
 	}()
 
-	return ate.txSimulator.ProcessTx(tx)
+	currentHeader := ate.blockChain.GetCurrentBlockHeader()
```

Comment:
you need to treat the situation when the currentHeader is nil and take the genesis block. Otherwise the simulation won't work on the exact first block after the genesis :)

Reply:
fixed